intelmq/bots/experts/securitytxt/expert.py

```diff
@@ -0,0 +1,104 @@
+from typing import Optional
+
+import requests
+from securitytxt import SecurityTXT
+
+from intelmq.lib.bot import ExpertBot
+
+
+class SecurityTXTExpertBot(ExpertBot):
+    """
+    A bot for retrieving contact details from a security.txt
+    """
+    """
+    url_field: The field where to find the url which should be searched
+    contact_field: Field in which to place the found contact details
+
+    only_email_address: whether to select only email addresses as contact detail (no web urls)
+    overwrite: whether to override existing data
+    check_expired / check_canonical: whether to perform checks on expiry date / canonical urls.
+    """
+    url_field: str = "source.reverse_dns"
+    contact_field: str = "source.abuse_contact"
+
+    only_email_address: bool = True
+    overwrite: bool = True
+    check_expired: bool = False
+    check_canonical: bool = False
+
+    def init(self):
+        if not self.url_field or not self.contact_field:
+            raise AttributeError("Not all required fields are set.")
+
+    def process(self):
+        event = self.receive_message()
+
+        try:
+            self.check_prerequisites(event)
+            primary_contact = self.get_primary_contact(event.get(self.url_field))
+            event.add(self.contact_field, primary_contact, overwrite=self.overwrite)
+        except NotMeetsRequirementsError as e:
+            self.logger.debug(str(e) + " Skipping event.")
+        except ContactNotFoundError as e:
+            self.logger.debug(f"No contact found. {str(e)} Continue.")
```

Comment:
```suggestion
            self.logger.debug("No contact found: %s Continue.", e)
```
Moving the formatting step to the logging facility has the advantage that if the formatting fails for some reason, the message is still being logged and no additional exception is thrown.